### DISCUSSION TITLE
webhook authentication

### DISCUSSION THREAD

**Discussion by schmurfy** [concourse/concourse#6172] [Help & Support] [answered]
I tried to setup webhook for a resource with github but all I get is a 401, is authentication enabled on webhooks ?
I tried to send the query myself but even with a user/pass sent as basic or digest auth it returns a 401

**Comment by skreddy6673** [comment]
Can you provide the webhook payload and the resource pipeline config? You can mask sensitive information if you have any.

**Comment by skreddy6673** [comment, marked as answer]
### Configuring a Git Resource

In your git resource, you will need to configure the resource with a `webhook_token`.  Here is a sample git resource with `check_every` and `webhook_token` set.  
```
  - name: source-repo
    type: git
    check_every: 8760h
    source:
      uri: ((URL))
      branch: master
    webhook_token: ThisIsMyWebhookToken
```

### Configure GitHub Webhook

To configure the GitHub webhook, you will need to be a repo administrator and have access to the repository settings.  Each webhook delivery will trigger the Concourse git resource to check if there is an update to the branch configured in the resource.  

1) Navigate to the repo on your github
2) Open the repository Settings
3) Click on the `Hooks` option
4) Click the `Add Webhook` button
5) Enter the details:
  - Payload URL: *this follows the [Concourse Resource Webhook Specification](https://concourse-ci.org/resources.html#resource-webhook-token) format.*  An example is `https://CONCOURSE_URL/api/v1/teams/YOUR_TEAM_NAME/pipelines/YOUR_PIPELINE_NAME/resources/YOUR_RESOURCE_NAME/check/webhook?webhook_token=ThisIsMyWebhookToken`
  - In place of `ThisIsMyWebhookToken` use some random string here
  - Content type: application/x-www-form-urlencoded
  - Webhook Secret: *leave this empty*
  - SSL Verification: Enable SSL Verification
  - Which events: *this is up to you but at least do the push event*

Once you create the webhook, push some changes to the github repo and see your resource in pipeline checked automatically

**Reply by schmurfy** [reply]
Hello, I carefully followed again the steps and it now works... I feel really stupid xD
I think I mixed up the tokens for the two git sources I have.

Thanks for taking the time to write that really detailed step by step instructions, I feel like it should be part of the documentation since github is probably the more widely used git host.
